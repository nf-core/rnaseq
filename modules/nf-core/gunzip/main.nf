process GUNZIP {
    tag "$archive"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/ubuntu:20.04' :
        'nf-core/ubuntu:20.04' }"

    input:
    tuple val(meta), path(archive)

    output:
    tuple val(meta), path("$gunzip"), emit: gunzip
    path "versions.yml"             , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    gunzip = archive.toString() - '.gz'
    """
    # Not calling gunzip itself because it creates files
    # with the original group ownership rather than the
    # default one for that user / the work directory
    gzip \\
        -cd \\
        $args \\
        $archive \\
        > $gunzip

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        gunzip: \$(echo \$(gunzip --version 2>&1) | sed 's/^.*(gzip) //; s/ Copyright.*\$//')
    END_VERSIONS
    """

    stub:
    gunzip = archive.toString() - '.gz'
    """
    touch $gunzip
    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        gunzip: \$(echo \$(gunzip --version 2>&1) | sed 's/^.*(gzip) //; s/ Copyright.*\$//')
    END_VERSIONS
    """
}
