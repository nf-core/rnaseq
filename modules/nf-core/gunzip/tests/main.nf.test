nextflow_process {

    name "Test Process GUNZIP"
    script "../main.nf"
    process "GUNZIP"
    tag "gunzip"
    tag "modules_nfcore"
    tag "modules"

    test("Should run without failures") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = Channel.of([
                        [],
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkIfExists: true)
                    ]
                )
                """
            }
        }

        then {
            assertAll(
            { assert process.success },
            { assert snapshot(process.out).match() }
            )
        }

    }

}
