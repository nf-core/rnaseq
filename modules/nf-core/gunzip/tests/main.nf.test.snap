{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        [
                            
                        ],
                        "test_1.fastq:md5,4161df271f9bfcd25d5845a1e220dbec"
                    ]
                ],
                "1": [
                    "versions.yml:md5,54376d32aca20e937a4ec26dac228e84"
                ],
                "gunzip": [
                    [
                        [
                            
                        ],
                        "test_1.fastq:md5,4161df271f9bfcd25d5845a1e220dbec"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,54376d32aca20e937a4ec26dac228e84"
                ]
            }
        ],
        "timestamp": "2023-10-17T15:35:37.690477896"
    }
}